### ISSUE TITLE
Display of road names is somewhat unreliable

### ISSUE THREAD

**Issue by sjvudp** [westnordost/StreetComplete#5968]
Walking around to fill in missing data, I thought there was a road without a name, so I added one.
However after a while the name of the road had appeared, and obviously the name was there, but not displayed.

**How to Reproduce**

I wondered what might cause the effect, so I played a bit:
It seems rotating the map (e.g. with rotation following movement) makes some street names appear while others disappear.

I was able to reproduce (I tried hard not to change the zoom factor while rotation the map with two fingers):

<img src="https://github.com/user-attachments/assets/e286283b-40c2-47d2-af14-d6c9b1bd0dbd" width="40%"> <img src="https://github.com/user-attachments/assets/abc26a50-b75b-4f29-bfa0-6cfe5dd45a60" width="40%">

On the left screenshot "Amselweg" and "Elsterweg" are missing.

**Expected Behavior**

A more consistent display of street names when the map is rotated.

**Versions affected**

v59.2 on Android 13 (Oppo Color OS 13.1)

**Comment by HolgerJeromin** [comment]
Related or duplicate of #5902 ?

**Comment by sjvudp** [comment]
> Yes, duplicate of #5902 . The reason for this behavior is that when quest pins collide with name labels, the labels are not shown.

Can't the street names be displayed *under* the quest pins?

**Comment by westnordost** [comment]
Yes, but then they would not collide with anything, i.e. not with each other, resulting in a street name woed salad on lower zooms.

https://github.com/maplibre/maplibre-style-spec/issues/822

**Comment by sjvudp** [comment]
I must admit that I don't understand the details well enough,but when walking with the "read arrow" mode enabled (follow position and direction of movement) it's odd to walk along a road that has no name displayed when it actually has a name.
In that case the zoom is much higher than in the examples shown.

**Comment by westnordost** [comment]
> I must admit that I don't understand the details well enough

Labels and icons in MapLibre (the map rendering library we use now) can either always render or set to "collide" with other labels and icons when they overlap. Colliding will make MapLibre choose to only display one of those labels/icons that overlap with each other, the label/icon with higher priority. Now of course, map pins, being displayed on top of the map, have a higher priority than street labels. So whenever there would be a road label near a quest pin, the road label is not rendered.

With a map full of pins, then, it follows, that basically no label is rendered, because there is no space for it. A solution would be to be able to configure which labels/icons collide with which other labels/icons, so that the road labels can be configured to collide with each other, but not with map pins, and the map pins in turn can also to be configured to collide with each other but not with road labels. This solution is proposed in the linked issue.

**Comment by sjvudp** [comment]
OK, I wonder: Assume you don't see the street name, and thus decide to add a note (hint) containing the name.
Would it make sense to display some data (like house address or street name) around the point where the note is going to be placed? So one could avoid placing such a seemingly useless note.

**Comment by riQQ** [comment]
Blocked by maplibre/maplibre-style-spec#822
